app/models/piece.rb

```diff
@@ -1,4 +1,34 @@
 class Piece < ApplicationRecord
   belongs_to :player
   belongs_to :game
+
+  def obstructed?(destination_x, destination_y)
+    x_difference = x_coordinate - destination_x
+    y_difference = y_coordinate - destination_y
+
+    chess_in_between = []
+
+    # first, raise an argument error when self and sq have the same coordinates
+    if x_difference.zero? && y_difference.zero?
+      raise ArgumentError, 'destination has to have a different location'
```

Comment:
Saw rubocop mention this. Something you can do here is use a guard clause.

guard clause is basically an if on a single line that stops the method from executing.

```
raise ArgumentError, 'destination has to have a different location' if x_difference.zero? && y_difference.zero?

if x_difference.zero?
 ...etc  
```